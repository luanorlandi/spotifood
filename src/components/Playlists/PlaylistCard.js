import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';

import './PlaylistCard.scss';

const PlaylistCard = ({ playlist }) => {
  const { name, description } = playlist;
  const imageUrl = get(playlist, 'images[0].url');
  const spotifyLink = get(playlist, 'external_urls.spotify');

  return (
    <a href={spotifyLink} className="playlist-card">
      <img
        className="playlist-card__image"
        src={imageUrl}
        alt="Capa da playlist"
      />
      <div>
        <div className="playlist-card__title">{name}</div>
        <div>{description}</div>
      </div>
    </a>
  );
};

PlaylistCard.propTypes = {
  playlist: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    images: PropTypes.array,
    external_urls: PropTypes.object,
  }).isRequired,
};

export default PlaylistCard;
