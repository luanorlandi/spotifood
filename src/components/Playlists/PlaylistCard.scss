@import '~bulma/sass/utilities/_all.sass';

.playlist-card {
  padding: 16px;
  background-color: #252525;
  color: #b3b3b3;
  transition: 200ms;

  &:hover {
    color: #b3b3b3;
    background-color: #333333;
  }

  @include touch {
    text-align: center;
  }
}

.playlist-card__image {
  width: 300px;
  height: 268px;
}

.playlist-card__title {
  color: #fff;
  font-size: 1.1rem;
  font-weight: 700;
}