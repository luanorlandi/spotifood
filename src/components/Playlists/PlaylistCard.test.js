import React from 'react';
import { render } from '@testing-library/react';

import PlaylistCard from './PlaylistCard';

const playlist = {
  id: '37i9dQZF1DX4sWSpwq3Lia',
  name: 'Sleep',
  description: 'Gentle ambient piano to help you fall asleep.',
  images: [{ url: 'https://i.scdn.co/image/ab67706f00000002b70e0223f544b1faa2e95ed0' }],
  external_urls: {
    spotify: 'https://open.spotify.com/playlist/37i9dQZF1DWZd79rJ6a7lp',
  },
};

test('playlist card match snapshot', () => {
  const { asFragment } = render((
    <PlaylistCard playlist={playlist} />
  ));

  expect(asFragment()).toMatchSnapshot();
});

test('playlist card has a link', async () => {
  const { container } = render((
    <PlaylistCard playlist={playlist} />
  ));

  // not advisible to query by element
  // but in this case it's intended to assert that the card always has a link
  const linkElement = container.querySelector('a');
  expect(linkElement).toHaveAttribute('href', playlist.external_urls.spotify);
});
