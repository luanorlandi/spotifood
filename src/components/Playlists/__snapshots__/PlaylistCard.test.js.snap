// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playlist card match snapshot 1`] = `
<DocumentFragment>
  <a
    class="playlist-card"
    href="https://open.spotify.com/playlist/37i9dQZF1DWZd79rJ6a7lp"
  >
    <img
      alt="Capa da playlist"
      class="playlist-card__image"
      src="https://i.scdn.co/image/ab67706f00000002b70e0223f544b1faa2e95ed0"
    />
    <div>
      <div
        class="playlist-card__title"
      >
        Sleep
      </div>
      <div>
        Gentle ambient piano to help you fall asleep.
      </div>
    </div>
  </a>
</DocumentFragment>
`;
